lib/storage/storage.go

```diff
@@ -1927,7 +1930,7 @@ func (s *Storage) add(rows []rawRow, dstMrs []*MetricRow, mrs []MetricRow, preci
 
 		addToPendingHourEntries(hour, genTSID.TSID.MetricID)
 
-		if logNewSeries {
+		if logNewSeries || (s.logNewSeriesUntil.Load() >= fasttime.UnixTimestamp()) {
```

Comment:
I suggest to move this check on the top of `(s *Storage) add()` function:

```
logNewSeries := s.logNewSeries || s.logNewSeriesUntil.Load() >= fasttime.UnixTimestamp()
```

 And embed global variable `logNewSeries` into Storage struct.